internal/app/networkd/pkg/nic/nic.go

```diff
@@ -244,6 +248,16 @@ func (n *NetworkInterface) Configure() (err error) {
 		}
 	}
 
+	if n.VirtualIP != nil {
+		if n.vipController, err = vip.New(n.VirtualIP.String(), n.Link.Name); err != nil {
+			return fmt.Errorf("failed to create the VirtualIP controller for %q on %q: %w", n.VirtualIP, n.Link.Name, err)
+		}
+
+		if err = n.vipController.Start(context.TODO()); err != nil {
```

Comment:
with such context we don't have a way to shut down `vipController` gracefully, so when node goes down for a reboot, it doesn't look like etcd elections will be immediately retried to move VIP to a different node? and VIP is not technically released from the interface when networkd goes down?

Reply:
So the election is safe:  etcd leases are immediately expired when the client connection is lost.

However, you are absolutely correct that we should shut down properly.  I just didn't see any way to get a control context in here with the current networkd architecture.  That's why the context here is a TODO.   When networkd is reworked, it will have a control context which can be shut down gracefully.

Reply:
is it a problem about passing down a context? we could start around `main` function, initialize a context, cancel it on signal (service runner first sends SIGTERM) and push it through the code to reach to the parts liks this one :)

Reply:
Yeah, exactly.